backend/app/views.py

```diff
@@ -243,6 +244,43 @@ def get_corpus(request, corpus_id):
     return Response(serializer.data)
 
 
+@api_view(['GET'])
+def all_proximity(request):
+    prox_objs = ProximityAnalysis.objects.all()
+    serializer = ProximityAnalysisSerializer(prox_objs, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['POST'])
+def add_proximity_analysis(request):
+    """
+    API endpoint for posting the proximity analysis
+    """
+    attributes = request.data
+    corpus_id = int(attributes['corpus_id'])
+    word_window = int(attributes['word_window'])
+
+    proximity_query = ProximityAnalysis.objects.filter(corpus__id=corpus_id, word_window=word_window)
+
+    if proximity_query.exists():
+        proximity_obj = proximity_query.get()
+
+    else:
+        results = run_analysis(corpus_id, word_window)
+        fields = {
+            'corpus': Corpus.objects.get(pk=corpus_id),
+            'word_window': word_window,
+            'results': results,
+        }
+
```

Comment:
Ultimately, we may want to change this up so that you call a ProximityAnalysisManager or similar (like we do with the DocumentManager) when we create this new ProximityAnalysis, and have the ProximityAnalysisManager call the run_analysis function. That is: since a ProximityAnalysis necessarily relies on the run_analysis function, we could use the ProximityAnalysis itself (through a Manager) to ensure that run_analysis is run when a new instance is created. That would more tightly couple associated functionality. That said, since this is working and is well tested, we can leave it be for now and come back to it later.